Py_TYPE() is a function, not an assignable macro, since Python 3.10, so use
Py_SET_TYPE() where it is available. Boost.ParameterPython has not been
updated since 2018, see
https://github.com/boostorg/parameter_python/blob/develop/include/boost/parameter/python.hpp

--- a/include/boost/parameter/python.hpp
+++ b/include/boost/parameter/python.hpp
@@ -66,7 +66,11 @@
       
       if (Py_TYPE(&unspecified) == 0)
       {
+#if PY_VERSION_HEX >= 0x03090000
+          Py_SET_TYPE(&unspecified, &PyType_Type);
+#else
           Py_TYPE(&unspecified) = &PyType_Type;
+#endif
           PyType_Ready(&unspecified);
       }
       
